### DISCUSSION TITLE
Incorrect eccentricity data from API in RFEM 6.03

### DISCUSSION THREAD

**Discussion by sonkaste1** [Dlubal-Software/Dlubal_CSharp_Client#44] [General]
Hi,

in versions RFEM 6.03 (specifically we tested 6.03.0005 and 6.03.0007) we found out, that if we use the API call for getting the eccentricity data (get_member_eccentricity(id)) we receive object with correct name, but the values are always empty (all flags set to false and values are zero). 
With the same code in version 6.02.0065 and previous ones we got correct data without any issues.

Here is attached screenshot from debugging with version 6.03:

<img width="875" alt="MicrosoftTeams-image (2)" src="https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/d33a8858-a609-4d97-a313-d100954d6c61">

Here is screenshot of the object in RFEM model:

<img width="572" alt="MicrosoftTeams-image (3)" src="https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/f40d5ed5-1763-4f4a-bbc9-723683896df1">

Thank you for help!

**Comment by Rebecca-Coding21** [comment]
Hi @sonkaste1,
I couldn't reproduce the problem in the current version. Please make sure that you are using the current NuGet package with version 6.3.6:
![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/94243632/d6e9aa7f-173f-44fe-89d0-792b4e3f15ef)

**Reply by sonkaste1** [reply]
@Rebecca-Coding21 Thank you! The latest NuGet solved the issue.

**Comment by sonkaste1** [comment]
Hi @Rebecca-Coding21 ,

during test round we found out, that if we use the recommended latest Nuget package version 6.3.6, it is working correctly for RFEM 6.03, but it is not working correctly with the RFEM 6.02 (we tested 6.02.0065). There is completely empty object of eccentricity, this time even without name:

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/4b469682-7919-42f8-a2d7-4df69b0bbce0)

Is it a bug or does it mean, that there is no backwards compatibility between Nugets and versions of RFEM? Are there any public release notes related to Nugets, where we could find, what is new/fixed in specific version please?

![image](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/87471266/5773c2fd-150c-4e66-ab8b-9ad94d8971a4)
 
Thanks for your answer.

**Comment by Rebecca-Coding21** [comment]
Hi @sonkaste1,
backward compatibility can't be guaranteed, especially between the "big" versions (6.02/6.03). The C# library is automatically generated from  the main program. It is possible that the functions in the main program changed between version 6.02 and 6.03. With the older NuGet package 6.2.67 the eccentricity function should work with RFEM 6.02.0065. There shouldn't be any problems with the API between the weekly versions, but it is of course recommended to always use the latest/corresponding versions of the NuGet package and RFEM.

You can find the release under the following link:
[Release XX.03.007](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/discussions/45)

Because of the generated code you will not find detailed information there about bug fixes or new features. These can be found in the Dlubal Extranet unter "Development":
![](https://github.com/Dlubal-Software/Dlubal_CSharp_Client/assets/94243632/c5076b4c-16bb-44fc-96be-4607c4aa9e94)
